tests/testsuite/script.rs

```diff
@@ -306,20 +309,25 @@ fn use_script_config() {
         .at("script")
         .file("script.rs", script)
         .build();
+    let root = paths::root();
+    let my_home = root.join("my_home");
+    fs::create_dir(&my_home).unwrap();
+    fs::write(
+        &my_home.join("config.toml"),
+        r#"
+            [build]
+            rustc = "non-existent-rustc"
+        "#,
+    )
+    .unwrap();
 
     let p = cargo_test_support::project()
-        .file(
-            ".cargo/config.toml",
-            r#"
-[build]
-rustc = "non-existent-rustc"
-"#,
-        )
         .file("script.rs", script)
         .build();
 
     // Verify the config is bad
     p.cargo("-Zscript script.rs -NotAnArg")
+        .env("CARGO_HOME", &my_home)
```

Comment:
Why are we overriding CARGO_HOME rather than using the cargo home created for you?

Reply:
See https://doc.rust-lang.org/nightly/nightly-rustc/cargo_test_support/paths/fn.cargo_home.html

Reply:
I don't want the second case to be affected by the default cargo home either.

Reply:
I believe I should change the test case to focus solely on checking the cargo home configuration. I will update it.

Reply:
Updated. 
1. I checked that the config from the current directory is used.
2. I checked that the config from the parent directory is not used.
3. I checked that config from the home directory is used.
4. Renamed it to use_cargo_home_config.